sdk/tables/azure-data-tables/azure/data/tables/_table_service_client.py

```diff
@@ -0,0 +1,295 @@
+        :rtype: None
+        """
+        _validate_table_name(table_name)
+
+        self._client.table.delete(table=table_name, **kwargs)
+
+    @distributed_trace
+    def query_tables(
+            self,
+            filter,  # pylint: disable=W0622
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> ItemPaged[Table]
+        """Queries tables under the given account.
+        :param filter: Specify a filter to return certain tables
+        :type filter: str
+        :keyword int results_per_page: Number of tables per page in return ItemPaged
+        :keyword Union[str, list(str)] select: Specify desired properties of a table to return certain tables
+        :keyword dict parameters: Dictionary for formatting query with additional, user defined parameters
+        :return: A query of tables
+        :rtype: ItemPaged[Table]
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        parameters = kwargs.pop('parameters', None)
+        if parameters:
+            selected = filter.split('@')[1]
+            for key, value in parameters.items():
+                if key == selected:
+                    filter = filter.split('@')[0].replace('@', value)  # pylint: disable = W0622
+
+        temp_select = kwargs.pop('select', None)
+        select = " "
+        if temp_select is not None and len(list(temp_select)) > 1:
+            select = ", ".join(temp_select)
+
+        query_options = QueryOptions(top=kwargs.pop('results_per_page', None), select=select or temp_select,
+                                     filter=filter)
+        command = functools.partial(self._client.table.query,
+                                    **kwargs)
+        return ItemPaged(
+            command, results_per_page=query_options,
+            page_iterator_class=TablePropertiesPaged
+        )
+
+    @distributed_trace
+    def list_tables(
+            self,
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> ItemPaged[Table]
+        """Queries tables under the given account.
+
+        :keyword int results_per_page: Number of tables per page in return ItemPaged
+        :keyword Union[str, list(str)] select: Specify desired properties of a table to return certain tables
+        :keyword dict parameters: Dictionary for formatting query with additional, user defined parameters
+        :keyword str filter: Specify desired filter for tables
+        :return: A query of tables
+        :rtype: ItemPaged[Table]
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        parameters = kwargs.pop('parameters', None)
+        filter = kwargs.pop('filter', None)  # pylint: disable=W0622
+        if parameters:
+            filter_start = filter.split('@')[0]
+            selected = filter.split('@')[1]
+            for key, value in parameters.items():
+                if key == selected:
+                    filter = filter_start.replace('@', value)  # pylint: disable = W0622
```

Comment:
We don't support `filter` in the list operation - so we can remove this :)
However in general I think we should move the parameter substitution code out into a separate function.